core/finality-grandpa/src/lib.rs

```diff
@@ -1524,6 +1533,7 @@ pub fn run_grandpa<B, E, Block: BlockT<Hash=H256>, N, RA>(
 		);
 
 		let committer_data = committer_communication(
+			config.local_key.clone(),
```

Comment:
we might rather want to move the logic for checking if we are a current voter to just above this function call. Then the `outgoing_messages` for every round will also be a no-op.

```
config.local_key = config.local_key.filter(|pair| voters.contains_key(&pair.public().into()))
```